File: 2020/day3.py
INPUT_FILE = 'inputs/day3_input.txt'

def load_data():
    with open(INPUT_FILE, 'r') as f:
        data = f.readlines()
    data = [x.strip() for x in data]
    return data

def toboggan(forest, slope=(1,3)):
    row=0
    col=0
    trees=0
    while row < len(forest)-1:
        print(f"Row: {row}, col: {col}, trees: {trees}")
        row+=slope[0]
        col+=slope[1]
        if col >= len(forest[row]):
            col-=len(forest[row])
        if forest[row][col] == '#':
            trees+=1
    return trees
        
def main():
    data = load_data()
    trees = toboggan(data)
    print(trees)

def main_part2():
    data = load_data()
    slopes = [(1,1),(1,3),(1,5),(1,7),(2,1)]
    tree_mult = 1
    for slope in slopes:
        trees = toboggan(data, slope)
        tree_mult*=trees
    print(tree_mult)

if __name__=="__main__":
    main_part2()

File: 2020/julia/day3.jl
INPUT_FILE = "inputs/day3_input.txt"

function load_data()
    data = readlines(INPUT_FILE)
    data = [strip(x) for x in data]
end

function toboggan(forest, slope=(1,3))
    row=1
    col=1
    trees=0
    while row < length(forest)
        println("Row: $row, col: $col, trees: $trees")
        row+=slope[1]
        col+=slope[2]
        if (col > length(forest[row]))
            col-=length(forest[row])
        end
        if forest[row][col] == '#'
            trees+=1
        end
    end
    return trees
end

function part1()
    data = load_data()
    trees = toboggan(data)
    println("##### DAY ONE RESULT #####")
    println(trees)
end

function part2()
    data = load_data()
    slopes = [(1,1),(1,3),(1,5),(1,7),(2,1)]
    trees_mult = 1
    for slope in slopes
        trees = toboggan(data, slope)
        trees_mult = trees_mult * trees
    end
    println("##### DAY TWO RESULT #####")
    println(trees_mult)
end

part2()